test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -324,6 +414,153 @@ private void checkTransactionReadAndWrite() throws TxnFailedException {
+        // Deleting key value table and validating it.
+        assertTrue(keyValueTableManager.deleteKeyValueTable(scopeName, keyValueTableName));
+    }
+
+    /**
+     * This method writes and reads large events (4MB) and validates them.
+     */
+    @Test(timeout = 20000)
+    private void checkWriteAndReadLargeEvent() {
+        String scopeName = "largeevent-test-scope";
+        String streamName = "largeevent-test-stream";
+        createScopeAndStream(scopeName, streamName);
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        EventStreamWriter<ByteBuffer> writer = clientFactory.createEventWriter(streamName, new ByteBufferSerializer(), EventWriterConfig.builder().enableLargeEvents(true).build());
+
+        byte[] payload = new byte[Serializer.MAX_EVENT_SIZE * 4];
+        for (int i = 0; i < NUM_EVENTS; i++) {
+            log.info("Writing event: {} ", i);
+            // any exceptions while writing the event will fail the test.
+            writer.writeEvent("", ByteBuffer.wrap(payload)).join();
+            log.info("Wrote event: {} ", i);
+            writer.flush();
+        }
+        String readerGroupId = getRandomID();
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder().stream(Stream.of(scopeName, streamName)).build();
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scopeName, controllerURI);
+        readerGroupManager.createReaderGroup(readerGroupId, readerGroupConfig);
+        @Cleanup
+        EventStreamReader<ByteBuffer> reader =  clientFactory.createReader("reader-1", readerGroupId, new ByteBufferSerializer(), ReaderConfig.builder().build());
+
+        int readCount = 0;
+        EventRead<ByteBuffer> event = null;
+        do {
+            event = reader.readNextEvent(10_000);
+            log.debug("Read event: {}.", event.getEvent());
+            if (event.getEvent() != null) {
+                readCount++;
+            }
+            // try reading until all the written events are read, else the test will timeout.
+        } while ((event.getEvent() != null || event.isCheckpoint()) && readCount < NUM_EVENTS);
+        assertEquals("Read count should be equal to write count", NUM_EVENTS, readCount);
+    }
+
+    @Data
+    private static class PaddedStringSerializer {
+        private final int maxLength;
+
+        ByteBuffer serialize(String s) {
+            Preconditions.checkArgument(s.length() <= maxLength);
+            s = Strings.padStart(s, this.maxLength, ' ');
+            return ByteBuffer.wrap(s.getBytes(StandardCharsets.US_ASCII));
+        }
+
+        String deserialize(ByteBuffer b) {
+            Preconditions.checkArgument(b.remaining() <= maxLength);
+            String s = StandardCharsets.US_ASCII.decode(b).toString();
+            s = s.trim();
+            return s;
```

Comment:
s can be removes as it is not used elsewhere. 
return statement can be changed to StandardCharsets.US_ASCII.decode(b).toString().trim();